src/Microsoft.AspNet.Identity/SignInManager.cs

```diff
@@ -75,6 +75,21 @@ public class SignInManager<TUser> where TUser : class
             return Logger.Log(true);
         }
 
+        /// <summary>
+        /// Called to regenerate the ApplicationCookie for the user, preserving the existing
+        /// AuthenticationProperties like rememberMe
+        /// </summary>
+        /// <param name="user"></param>
+        /// <returns></returns>
+        public virtual async Task ResignInAsync(TUser user)
+        {
+            var authResult = await Context.AuthenticateAsync(IdentityOptions.ApplicationCookieAuthenticationScheme);
+            var properties = authResult?.Properties;
+            var isPersistent = properties != null && properties.IsPersistent;
```

Comment:
@divega We can consider changing SignIn to mimic the underlying SignIn API more closely, remove IsPersistent and take in a raw AuthenticationProperties which would expose everything